### PR TITLE
Handle new DB or CV created by importer, add to collections

### LINKED ISSUE #2032: tripal_id_space_collection table missing terms in migrated Tripal 3 site
### Task Description

This came up in a slack discussion from @Ferrisx4 - For a migrated chado dump from a Tripal 3 site, when you go to "+Check for new fields", you may see an error similar to this
2024-11-16-invalidfields

The underlying reason is that this term Id Space, "NCBI_BioSample_Attributes" in the screenshot, is not present in the `tripal_id_space_collection` table. The suggestion from @lacey was "Things get added to the table when you create terms through that api. All of the terms used in the semweb mappings have already been created in this way and I think the same happens when you select a term for a field through the UI. I wonder if the property discover method should be creating a term using the api when it makes the field. That would solve this problem. ... maybe instead of fixing it in the property field it should get fixed in the load for tripal terms? that might be a harder fix but it would be a better one."

In the meantime, the workaround is to manually add any missing terms. For this example the SQL would be:
`INSERT INTO tripal_id_space_collection (name, plugin_id) VALUES ('NCBI_BioSample_Attributes', 'chado_id_space');`
`INSERT INTO tripal_vocabulary_collection (name, plugin_id) VALUES ('NCBI BioSample Attributes', 'chado_vocabulary');`

### Branch Name

tv4g1-issue2032-tripal_id_space_collection

### PR DESCRIPTION
**Pull request by dsenalik**
# Bug Fix

## Description
This adds automated handling of adding a DB or CV to its appropriate tripal ID space or vocabulary collection.
These are generated by importers when they encounter something new, but for tripal terms to work correctly, they need to be added to the `tripal_id_space_collection` table or `tripal_vocabulary_collection` table, respectively.
This is now done by `loadCollection()` in the `TripalCollectionPluginManager` class.
To do this, we need to pass the appropriate plugin ID from the discover() function, or from chado's getTerm().

## Testing?
1. Try the procedure described in this comment, it should work now
https://github.com/tripal/tripal/issues/2032#issuecomment-2525328676
2. Quick way to reproduce the importer bug, first create a study, you can use the null contact for this
3. Next, some SQL:
```
INSERT INTO DB (name) VALUES ('PR2054DB');
INSERT INTO CV (name) VALUES ('PR2054CV');
INSERT INTO dbxref (db_id, accession) VALUES ((SELECT db_id FROM db WHERE name='PR2054DB'), 'PR2054acc');
INSERT INTO cvterm (cv_id, name, dbxref_id) VALUES ((SELECT cv_id FROM cv WHERE name='PR2054CV'), 'PR2054cvterm', (SELECT dbxref_id FROM dbxref WHERE accession='PR2054acc'));
INSERT INTO studyprop (study_id, type_id, value) VALUES (1, (SELECT cvterm_id FROM cvterm WHERE name='PR2054cvterm'), 'PR2054prop');
```
5. Check for new terms on study. On 4.x you would see
PR2054 on4 x
On this branch you should see
PR2054 2

6. Verify the collections were added
```
sitedb=> select * from tripal_vocabulary_collection where name='iao';
 name |    plugin_id     
------+------------------
 iao  | chado_vocabulary
(1 row)

sitedb=> select * from tripal_vocabulary_collection where name='PR2054CV';
   name   |    plugin_id     
----------+------------------
 PR2054CV | chado_vocabulary
(1 row)

sitedb=> select * from tripal_id_space_collection where name='PR2054DB';
   name   |   plugin_id    
----------+----------------
 PR2054DB | chado_id_space
(1 row)
```

### CODE REVIEW

**Comment by dsenalik** [comment]
I just added the high priority label, because after importing a chado instance, before even publishing anything, I am seeing thousands of messages in the log of the form
```
 wid  | uid |  type  |                                               message                                                | variables | severity | link |                        location                         |                                                                                                                                                                                                                                          referer                                                                                                                                                                                                                                           |  hostname  | timestamp  
------+-----+--------+------------------------------------------------------------------------------------------------------+-----------+----------+------+---------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+------------
 3403 |   1 | tripal | ERROR: TripalTerm::setVocabulary(). The specified vocabulary, "molecular_function" does not exist.   | a:0:{}    |        3 |      | http://localhost:8400/cvterm/autocomplete/10?q=s        | http://localhost:8400/admin/structure/bio_data/manage/primer/add-field/tripal_entity/field_primer_name?destinations%5B0%5D%5Broute_name%5D=entity.tripal_entity.field_ui_fields&destinations%5B0%5D%5Broute_parameters%5D%5Bentity_type%5D=tripal_entity&destinations%5B0%5D%5Broute_parameters%5D%5Bfield_name%5D=field_primer_name&destinations%5B0%5D%5Broute_parameters%5D%5Btripal_entity_type%5D=primer&destinations%5B1%5D=/admin/structure/bio_data/manage/primer/fields/add-field | 172.17.0.1 | 1734186725
 3404 |   1 | tripal | ERROR: TripalTerm::setVocabulary(). The specified vocabulary, "biological_process" does not exist.   | a:0:{}    |        3 |      | http://localhost:8400/cvterm/autocomplete/10?q=s        | http://localhost:8400/admin/structure/bio_data/manage/primer/add-field/tripal_entity/field_primer_name?destinations%5B0%5D%5Broute_name%5D=entity.tripal_entity.field_ui_fields&destinations%5B0%5D%5Broute_parameters%5D%5Bentity_type%5D=tripal_entity&destinations%5B0%5D%5Broute_parameters%5D%5Bfield_name%5D=field_primer_name&destinations%5B0%5D%5Broute_parameters%5D%5Btripal_entity_type%5D=primer&destinations%5B1%5D=/admin/structure/bio_data/manage/primer/fields/add-field | 172.17.0.1 | 1734186725
 8309 |   1 | tripal | ERROR: TripalTerm::setVocabulary(). The specified vocabulary, "biomaterial_property" does not exist. | a:0:{}    |        3 |      | http://localhost:8400/cvterm/autocomplete/10?q=organism | http://localhost:8400/admin/structure/bio_data/manage/primer/add-field/tripal_entity/field_organism_id?destinations%5B0%5D%5Broute_name%5D=entity.tripal_entity.field_ui_fields&destinations%5B0%5D%5Broute_parameters%5D%5Bentity_type%5D=tripal_entity&destinations%5B0%5D%5Broute_parameters%5D%5Bfield_name%5D=field_organism_id&destinations%5B0%5D%5Broute_parameters%5D%5Btripal_entity_type%5D=primer&destinations%5B1%5D=/admin/structure/bio_data/manage/primer/fields/add-field | 172.17.0.1 | 1734187797
```
and there are so many that the page times out. This is trying to add an organism field, or even to just create an organism through the GUI. Some of these also come from ajax.

**Review by laceysanderson — APPROVED** [review]
✅ code looks solid.
✅ Automated test is great ❤️ 
✅ Manual testing according to the instructions worked exactly as expected.
Screenshot 2024-12-17 at 11 02 55 AM

Screenshot 2024-12-17 at 11 03 04 AM
